src/components/common/layout/base.njk

```diff
@@ -163,4 +169,10 @@
             }
         }
     </script>
-{% endblock %}
\ No newline at end of file
+
+    <script type="text/javascript"  {% if scriptNonce %} nonce="{{ scriptNonce }}"{%  endif %}>
```

Comment:
Thanks for the suggestion ! I see that I can use the htmlClasses variable to add a class to the html element, but I won’t be able to add an ID. I wanted to use an id rather than a class, to avoid the use of `!important` keyword to ensure that my styling, would override the styling set by the`govuk-template` class which was causing the issue of the grey background.

Reply:
Hi @SophiaDWokoma - my sense is that it would be possible to apply a class via the `htmlClasses` in Nunjucks and rely on the CSS cascade and specificity to ensure the desired background colour is applied (i.e. without reliance on `!important`). I _think_ the desired background colour would win simply by adding this to `application.scss`
```scss
.govuk-template__mobile {
  background-color: $govuk-body-background-colour;
} 
```
Are you getting a different result? I'd be happy to help.

Reply:
Yeah it was working when I tested it but I used govuk-colour("white") directly , but that variable  `$govuk-body-background-colour` does the same thing so I will replace with that. I just wanted to confirm that the styling would always override using classes. Thank you

Reply:
Did you have to use `!important`? I'm just wondering about what you said about needing to introduce an ID.

Reply:
I have tested again, and it looks like I will need the `!important` keyword. It is a bit flaky, in that sometimes it works and sometimes it doesn't .

![image](https://github.com/user-attachments/assets/49adc535-d01b-4849-bd5b-659edd5a172b)
![image](https://github.com/user-attachments/assets/e67d4637-6568-4db9-9d25-9d89bdb6927b)
![image](https://github.com/user-attachments/assets/cf9c9764-d665-4a3d-827f-9f5b8d0daa20)

Reply:
Right after, I tested it again without the !important and it worked but yeah not sure it will always be consistent
![image](https://github.com/user-attachments/assets/06bfca77-9c75-4cc2-9256-6a94aeb5b36a)

Reply:
Hi @SophiaDWokoma - sorry to hear you're encountering this problem. 

I'm not necessarily opposed to using `!important` if we have to, it was more the use of client-side JavaScript to add an `id` that I'd definitely want to avoid. That said, I wouldn't have thought `!important` would be necessary because, where competing declarations in the cascade layer with precedence are equal, the last applicable rule will apply.

In this case: 

* the `background-color: #f3f2f1` appears in a rule with a specificity score of `0,1,0` that is declared on line 1390 of `style.css` 
* the `background-color: #ffffff;` appears in a rule with a specificity score of `0,1,0` that is declared on 8084 of `style.css`

I've tried to reproduce what you've shared in the screenshots and I'm seeing the `#ffffff` colour overrides the earlier declaration as expected (screenshot below). 

<details>
<summary>Screenshot</summary>

This screenshot shows my IDE (with the relevant rules and Nunjucks template) on the right alongside the browser on the right (showing the screen and applied styles). 
<img width="1920" alt="Screenshot 2024-10-21 at 08 56 47" src="https://github.com/user-attachments/assets/c89b6544-7f5c-4349-87d4-5da6ba0006e1">

</details> 

Is it possible the Sass hadn't compiled when you were testing?

Reply:
Yeah I have tested and also see the same. So we won't need to use `!important`

<img width="1512" alt="Screenshot 2024-10-21 at 11 09 00" src="https://github.com/user-attachments/assets/d24e9d07-640e-462d-9c5a-bdc5b3610d58">
<img width="1512" alt="Screenshot 2024-10-21 at 11 09 12" src="https://github.com/user-attachments/assets/fc2c54ce-0bdb-42ec-af4e-1be9b38ce09f">